dissect/target/plugin.py

```diff
@@ -960,8 +960,8 @@ def _filter_compatible(
                 continue
             raise
 
-        if plugincls not in seen:
-            seen.add(plugincls)
+        if descriptor.path not in seen:
```

Comment:
You can also keep two sets of classes, one of compatible and one of incompatible. That way you can still do a shortcut for both paths (immediately yield if in compatible, skip if in incompatible).